### ISSUE TITLE
KI/KL10: Run TYMCOM-X

### ISSUE THREAD

**Issue by larsbrinkhoff** [rcornwell/sims#240]
Brainstorming here.

**Comment by larsbrinkhoff** [comment]
https://wiki.livingcomputers.org/doku.php?id=tymcom-x-simh describes @joeinwap bringup on SIMH/KS10.

**Comment by larsbrinkhoff** [comment]
The file (M33)BOOTKI.DOC is interesting:

```
                         TYMCOM-X PAPER-TAPE BOOTSTRAPS
                         ------------------------------


((INTRODUCTION))

     A bootstrap or IPL (Initial Program Load) is a  technique  of  loading  the
monitor  or  a  "stand-alone"  program  into  the  TYMCOM-X  by means of certain
preliminary instructions to run programs and read data.  This technique is  used
with  a  paper  tape  "bootstrap"  when the computer is brought up "cold" (e.g.,
after a maintenance diagnostic has run "stand-alone" on the system).

     This document describes how to create either single or multiple  copies  of
the two bootstrap paper tapes which could be used to load the TYMCOM-X operating
system:  BOTLOD which bootstraps from disk and FTALOD which  bootstraps  from  a
magnetic  tape.  Once the tape has been created, it is mounted in the PTR (Paper
Tape Reader) on the KI console;  the READ-IN DEVICE switches must be set to  340
(switches  5 and 9);  to read the bootstrap tape the system must be down and you
must hit the STOP, RESET and READIN keys.   (NOTE  -  use  of  these  tapes  are
documented in the manual bringup section of this document)

**Comment by larsbrinkhoff** [comment]
(M33)FTALOD.DOC

```
              FTALOD - BOOTSTRAPPING FROM A MONITOR BACKUP TAPE
               -------------------------------------------------


((INTRODUCTION))

     The purpose of FTALOD is to load a  monitor  from  a  mag-  tape  which  is
mounted  on  one  of  the AMPEX tape units.  This provides us with an additional
method of bringing up either  the  current,  field  test  or  old  monitor  from
mag-tape  (in  this  regard,  always clear any monitor version changes with your
supervisor).

     FTALOD is needed because the PDP-10 hardware read-in feature does not  work
with  the  IBM  type  of tape units.  Instead, a specially prepared FTALOD paper
tape must be mounted in the PTR  (Paper  Tape  Reader)  and  the  READIN  DEVICE
switches set to 104 (switch 5 and 9 on).

     The monitor magtape may be mounted on any of the units  on  the  1st  AMPEX
tape kontroller (i.e., on FTA0 through fta7).

     Then as described below, FTALOD can be used to load the monitor from tape.

                   PREPARATION OF THE MAGTAPE USED BY FTALOD
                   -----------------------------------------

     7 track magtapes are prepared for  FTALOD  just  as  they  are  for  normal
read-in  from DEC magtape units:  On 7 track drives MAGRIM is written at 556 bpi
and the monitor sav file is written at 800 BPI.  9 track tapes may  have  MAGRIM
and  the  monitor  sav file at any densities that the particular 9 track unit is
capable of handling.

     Consult the document on CREATING A MONITOR TAPE contained in this manual.

(NOTE - FTALOD actually does not use MAGRIM, it just skips over it;   MAGRIM  is
only  required  so that the same tapes may be used for read-in from both the DEC
and AMPEX magtape units, and so that two procedures are not needed for  creating
read-in tapes.).

     You are also referred to the document on TYMCOM-X PAPER-TAPE BOOTSTRAPS (in
this manual) for the techniques used for creating an FTALOD papertape.

                     USING THE FTALOD INITIAL PROGRAM LOAD
                     -------------------------------------

1.  If the system is up, follow the usual procedures to take it down.

2.  Mounting the magtape on a unit with the correct track/density is  important.
The  unit must be 7 or 9 track, depending on whether the tape was written on a 7
or 9 track unit, respectively.  Note that 7 track units expect tapes with MAGRIM
at  556  bpi  and  the  monitor  'SAV' file at 800 bpi, while 9 track units will
accept MAGRIM and the monitor sav file at any densities that the unit is capable
of reading.

3.  Set the console data switches as follows:  The left half (LH)  of  the  data
switches  (0  thru  17)  are  set  to ones, and the right half (18 thru 35) to a
number from 0 to 7 inclusive  reflecting  which  drive  number  is  being  used.
FTALOD  will  assume  that  the  FTALOD  tape  is  on  the  unit  ftan  where  n
=rh(switches);  For example, if the switches are set to 777777,,000003 the  unit
acccessed by this procedure will be fta3.

4.  Put the papertape of FTALOD into the paper tape reader (PTR).

5.  set the readin device select switches to the ptr (104 - 5 and 9 on).

6.  depress the STOP, RESET and READIN switches.

7.  If the papertape does not load properly, reinsert it and try again.

8.  At this point, if all is well, FTALOD will rewind the mag- tape,  skip  over
MAGRIM,  and  read the monitor into core;  at successful completion, FTALOD will
have positioned the tape after the monitor file's eof and started the monitor.

**Comment by larsbrinkhoff** [comment]
Tymcom-X monitor documentation:  
https://github.com/TYMCOM-X/169273.tape/tree/abc68e373db6be0104efe986d23624462ef691b9/mondoc

**Comment by larsbrinkhoff** [comment]
Maybe the EXEC:  
https://github.com/TYMCOM-X/169271_1.tape/tree/90fd82317d5e214159281a269f045c0e7af2df87/execx

**Comment by larsbrinkhoff** [comment]
This depends on #12.
